### DISCUSSION TITLE
How to solve the problem of page jitter

### DISCUSSION THREAD

**Discussion by haoynmail** [vikejs/vike#920] [Discussion]
I used the example in the react-router-dom router library (see attachment data-router.zip
[data-router.zip](https://github.com/brillout/vite-plugin-ssr/files/11582029/data-router.zip)
[data-router-VPS.zip](https://github.com/brillout/vite-plugin-ssr/files/11582030/data-router-VPS.zip)
) and implemented it using the built-in router in vite-plugin-ssr (see attachment data-router-VPS.zip), where the nested layout in the Todos page arise page jitter during addition and deletion operations. Please help.

**Comment by brillout** [comment]
It may be related to the fact that with React Router pages are loaded with a dynamic `import()`.

FYI https://vite-plugin-ssr.com/vue-router-and-react-router. Bottom line being:

> If you want a feature provided by Vue Router or React Router that vite-plugin-ssr is missing, then open a new feature request.

**Reply by haoynmail** [reply]
I would like to replace the react-router-dom router library with the built-in router of vite-plugin-ssr. In the example of react-router-dom, dynamic import() were not used, and there was no page jitter during add and delete operations. However, using the built-in router of vite-plugin-ssr resulted in page jitter during add and delete operations. Additionally, I attempted to use dynamic import in vite-plugin-ssr, but the issue was not resolved.

**Reply by brillout** [reply]
> using the built-in router of vite-plugin-ssr resulted in page jitter 

Minimal reproduction welcome and I'll have a look at it.

**Reply by haoynmail** [reply]
It is actually a nested layout problem, and using vite-plugin-ssr causes jitter in the nested layout region
![nested-layout](https://github.com/brillout/vite-plugin-ssr/assets/73579327/5f7805c9-b293-4016-a1bd-ffcb22f81dbe)

**Reply by brillout** [reply]
> vite-plugin-ssr causes jitter 

I don't think it's caused by vite-plugin-ssr. But I'm happy to be shown wrong by a minimal reproduction that doesn't use React Router.

**Reply by haoynmail** [reply]
I'm not saying this is caused by vite-plugin-ssr. What I mean is how to solve the page jitter. Please help.
